services/meta/client.go

```diff
@@ -957,6 +833,15 @@ func (c *Client) WaitForDataChanged() chan struct{} {
 	return c.changed
 }
 
+// commit assumes it is under a full lock
+func (c *Client) commit(data *Data) {
+	data.Index++
+	c.cacheData = data
+	c.Snapshot()
```

Comment:
error is ignored.  Should be returned if we can't save.  Also, should probably try to snapshot first and then update in-memory state.